### ISSUE TITLE
problem with fake-gato data

### ISSUE THREAD

**Issue by rangerek** [naofireblade/homebridge-weather-plus#48]
here is the log - looks like it finds **undefined** somwhere :( - but still saves some data in json:

```
[1/15/2019, 1:55:06 PM] [WeatherPlus] **Fakegato-timer: executeCallbacks**
[1/15/2019, 1:55:06 PM] [WeatherPlus] First entry undefined: 0
[1/15/2019, 1:55:06 PM] [WeatherPlus] Last entry undefined: 18
[1/15/2019, 1:55:06 PM] [WeatherPlus] Used memory undefined: 18
[1/15/2019, 1:55:06 PM] [WeatherPlus] 116 undefined: b734000000000000f3daed2103 0102 0202 03021900c00f00000000000000000101
[1/15/2019, 1:55:06 PM] [WeatherPlus] ** Fakegato-storage write FS file: /root/.homebridge/Homebridge_undefined_persist.json "firstEntry":0,"lastEntry":24,"usedMemory":24,"refTime":569236211,"initialTime":
[1/15/2019, 1:55:06 PM] [WeatherPlus] **Fakegato-timer: emptyData ** undefined
[1/15/2019, 1:59:04 PM] [WeatherPlus] Clock adjust undefined: 9810ee21
[1/15/2019, 1:59:04 PM] [WeatherPlus] Data request undefined: 01140500000000
[1/15/2019, 1:59:04 PM] [WeatherPlus] Address requested undefined: 5
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 6, Address: 6
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 7, Address: 7
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 8, Address: 8
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 9, Address: 9
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 10, Address: 10
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 11, Address: 11
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 12, Address: 12
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 13, Address: 13
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 14, Address: 14
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 15, Address: 15
[1/15/2019, 1:59:04 PM] [WeatherPlus] Data undefined:  1505000000 0100 0000 81f3daed210000 0000 00 0000 10 060000006009000007a6ffd0206a27 10 07000000b80b000007a6ffd0206a27 10 08000000100e000007b0ff34216027 10 090000006810000007baff34216027 10 0a000000c012000007c4ff98216027 10 0b0000001815000007cefffc216027 10 0c0000007017000007baff1c256027 10 0d000000c819000007c4ff1c256027 10 0e000000201c000007ceff1c256027 10 0f000000781e000007e2ff1c255627
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 16, Address: 16
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 17, Address: 17
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 18, Address: 18
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 19, Address: 19
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 20, Address: 20
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 21, Address: 21
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 22, Address: 22
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 23, Address: 23
[1/15/2019, 1:59:04 PM] [WeatherPlus] undefined Entry: 24, Address: 24
[1/15/2019, 1:59:04 PM] [WeatherPlus] Data undefined:  10 10000000d020000007e2ff1c255627 10 110000002823000007f6ffb8245627 10 1200000080250000070000b8245627 10 13000000d827000007c4ff1c256027 10 14000000302a000007ceff1c255627 10 15000000882c000007e2ffb8245627 10 16000000e02e000007f6ff54245627 10 1700000038310000070000f0235627 10 18000000b734000007f1ff8c235627
```

content of json:
```
cat /root/.homebridge/Homebridge_undefined_persist.json
{"firstEntry":0,"lastEntry":32,"usedMemory":32,"refTime":569236211,"initialTime":1547543411,"history":["noValue",{"time":1547543411,"setRefTime":1},{"temp":-1.6,"pressure":1009,"humidity":87.5,"time":1547543411},{"temp":-1.5,"pressure":1009,"humidity":88,"time":1547544011},{"temp":-1.4,"pressure":1009,"humidity":88,"time":1547544611},{"temp":-1.3,"pressure":1009,"humidity":88,"time":1547545211},{"temp":-0.9,"pressure":1009,"humidity":84,"time":1547545811},{"temp":-0.9,"pressure":1009,"humidity":84,"time":1547546411},{"temp":-0.8,"pressure":1008,"humidity":85,"time":1547547011},{"temp":-0.7,"pressure":1008,"humidity":85,"time":1547547611},{"temp":-0.6,"pressure":1008,"humidity":86,"time":1547548211},{"temp":-0.5,"pressure":1008,"humidity":87,"time":1547548811},{"temp":-0.7,"pressure":1008,"humidity":95,"time":1547549411},{"temp":-0.6,"pressure":1008,"humidity":95,"time":1547550011},{"temp":-0.5,"pressure":1008,"humidity":95,"time":1547550611},{"temp":-0.3,"pressure":1007,"humidity":95,"time":1547551211},{"temp":-0.3,"pressure":1007,"humidity":95,"time":1547551811},{"temp":-0.1,"pressure":1007,"humidity":94,"time":1547552411},{"temp":0,"pressure":1007,"humidity":94,"time":1547553011},{"temp":-0.6,"pressure":1008,"humidity":95,"time":1547553611},{"temp":-0.5,"pressure":1007,"humidity":95,"time":1547554211},{"temp":-0.3,"pressure":1007,"humidity":94,"time":1547554811},{"temp":-0.1,"pressure":1007,"humidity":93,"time":1547555411},{"temp":0,"pressure":1007,"humidity":92,"time":1547556011},{"temp":-0.15,"pressure":1007,"humidity":91,"time":1547556906},{"temp":-0.2,"pressure":1007,"humidity":90,"time":1547557506},{"temp":-0.1,"pressure":1007,"humidity":89,"time":1547558106},{"temp":0,"pressure":1007,"humidity":89,"time":1547558706},{"temp":0.1,"pressure":1007,"humidity":88,"time":1547559306},{"temp":-0.7,"pressure":1007,"humidity":97,"time":1547559906},{"temp":-0.5,"pressure":1007,"humidity":96,"time":1547560506},{"temp":-0.4,"pressure":1007,"humidity":95,"time":1547561106},{"temp":-0.2,"pressure":1007,"humidity":94,"time":1547561706}]}
```

found another part of a log maybe more helpful in diagnosing what is wrong :) 

```
[1/16/2019, 9:57:25 AM] [WeatherPlus] Initializing WeatherPlus platform...
[1/16/2019, 9:57:25 AM] [WeatherPlus] ** Fakegato-storage AddWriter : undefined
[1/16/2019, 9:57:25 AM] [WeatherPlus] Loading...
[1/16/2019, 9:57:26 AM] [WeatherPlus] ** Fakegato-storage read FS file: /root/.homebridge/Homebridge_undefined_persist.json
[1/16/2019, 9:57:26 AM] [WeatherPlus] ** Fakegato-timer Subscription : undefined
[1/16/2019, 9:57:26 AM] [WeatherPlus] Registring Events undefined
[1/16/2019, 9:57:26 AM] [WeatherPlus] Accessory undefined
[1/16/2019, 9:57:26 AM] [WeatherPlus] Initializing platform accessory 'Now'...
[1/16/2019, 9:57:26 AM] [WeatherPlus] Initializing platform accessory 'Today'...
[1/16/2019, 9:57:26 AM] [WeatherPlus] Initializing platform accessory 'In 1 Day'...
[1/16/2019, 9:57:26 AM] [WeatherPlus] Initializing platform accessory 'In 2 Days'...
[1/16/2019, 9:57:26 AM] [WeatherPlus] Initializing platform accessory 'In 3 Days'...
[1/16/2019, 9:57:26 AM] [WeatherPlus] Initializing platform accessory 'In 4 Days'...
[1/16/2019, 9:57:26 AM] [WeatherPlus] Initializing platform accessory 'In 5 Days'...
[1/16/2019, 9:57:26 AM] [WeatherPlus] Initializing platform accessory 'In 6 Days'...
```

**Comment by simont77** [comment]
The problem is that no accessory name was passed to fakegato. It will be fixed with the pull request actually under review.

**Comment by naofireblade** [comment]
I released version 2.5.0 with the fix from @simont77 .
